### ISSUE TITLE
[BUG] ScrollView in Pager causes partially shown views

### ISSUE THREAD

**Issue by mohammad-rahchamani** [fermoya/SwiftUIPager#74]
I'm trying to show multiple full-width pages that are scrollable vertically. 
everything is working fine, but when you scroll vertically, you'll see some portion of neighboring views.

```
struct ContentView: View {
    @State var pageIndex = 0
    
    var items: [PagerData] = [
        PagerData(id: 0),
        PagerData(id: 1),
        PagerData(id: 2),
        PagerData(id: 3),
        PagerData(id: 4),
        PagerData(id: 5),
        PagerData(id: 6),
        PagerData(id: 7),
        PagerData(id: 8),
        PagerData(id: 9),
        PagerData(id: 10),
    ]
    
    func pageView(_ item: PagerData) -> some View {
        ScrollView {
            VStack {
                Color.red
                    .frame(height: 300)
                Text("item \(item.id)")
                Color.green
                    .frame(height: 300)
                Text("item \(item.id)")
                Color.blue
                    .frame(height: 300)
            }
        }
    }


    var body: some View {

        Pager(page: self.$pageIndex,
              data: self.items,
              content: { item in
              self.pageView(item)
        })

    }

}

struct PagerData: Equatable, Identifiable {
    var id: Int    
}

```
I don't have this issue when I use a `UIScorllView` inside a `UIPageViewController`.

here is a demo:

![out1](https://user-images.githubusercontent.com/6257496/87851954-fa0d6e80-c912-11ea-87e4-294a11d6450c.gif)

OSX: macOS BigSur 11.0 Beta 
XCode: 12.0 Beta
iOS: 14.0 Beta
iPhone 8

**Comment by fermoya** [comment]
@mohammad-rahchamani , I'm running your code on both Xcode 11.5 and Xcode 12 Beta 2 and I don't see this issue you mention. See iPhone 8 - 13.5 on the left and iPhone 8 - 14.0 on the right:
![movie](https://user-images.githubusercontent.com/11335612/87852336-f2d96700-c8f8-11ea-873b-07cadb21af9d.gif)

[movie.mov.zip](https://github.com/fermoya/SwiftUIPager/files/4941614/movie.mov.zip)

**Comment by mohammad-rahchamani** [comment]
Thank you for your fast response.
You have to drag diagonally for this to happen.
I'm trying to capture another demo with my pointer for more clarification.

**Comment by mohammad-rahchamani** [comment]
here is another demo with a visible gesture.

![demo1](https://user-images.githubusercontent.com/6257496/87852869-faaa0300-c91a-11ea-8c8d-0bd85028eef0.gif)

**Comment by fermoya** [comment]
@mohammad-rahchamani , I'm seeing it now and I can reproduce it, thanks. As far as I can see, the `DragGesture.onEnded` callback isn't called, `ScrollView` cancels the gesture. I've found a related issue, you can find it [here](https://stackoverflow.com/questions/60305207/interaction-of-draggesture-and-scrollview-in-swiftui). I can't really follow the same approach as `GestureState` resets before `onEnded` is called, so I wouldn't be able to animate the scroll to the next page.

I'll think about possible solutions, any thought on your side are more than welcome

**Comment by mohammad-rahchamani** [comment]
@fermoya please take a look at my pull request. 
I'm not able to reproduce the issue with this new change.

**Comment by carlosrc** [comment]
Same problem here!

**Comment by fermoya** [comment]
I’m working on it. I checked the PR but the issue kept happening.

I wanted to merge something first to avoid conflicts, I’ll be working on this promptly

**Comment by skt216** [comment]
Thank you. I am also facing the same issue

**Comment by fermoya** [comment]
I just released [v1.9.0-beta.4](https://github.com/fermoya/SwiftUIPager/releases/tag/1.9.0-beta.4) with a fix for this issue. You'll see that I'm ignoring swipes that aren't along the X-Axis ± 25 degrees.

Hope it works.

**Comment by carlosrc** [comment]
Thanks for the fix! Great work
